.github/workflows/github-template-update-downstream.yml

```diff
@@ -0,0 +1,102 @@
+        default: 'update-downstreams.yml'
+
+jobs:
+  get-repos:
+    runs-on: ubuntu-latest
+    outputs:
+      repos: ${{ steps.get-repos.outputs.repos }}
+    steps:
+      - name: Authenticate with GitHub App
+        id: auth
+        uses: tibdex/github-app-token@3beb63f4bd073e61482598c45c71c1019b59b73a # v2.1.0
+        with:
+          app_id: ${{ secrets.TERRAFORM_AUTOMATION_APP_ID }}
+          private_key: ${{ secrets.TERRAFORM_AUTOMATION_PRIVATE_KEY }}
+      
+      - name: Get repositories with prefix
+        id: get-repos
+        run: |
+          repos=$(gh repo list workleap --limit 1000 --json name --jq '[.[] | select(.name | startswith("terraform-")) | .name]')
+          echo "repos=$repos" >> $GITHUB_OUTPUT
+        env:
+          GH_TOKEN: ${{ steps.auth.outputs.token }}
+
+  sync:
+    needs: get-repos
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        repo: ${{ fromJson(needs.get-repos.outputs.repos) }}
+      fail-fast: false
+    steps:
+      - name: Authenticate
+        id: auth
+        uses: tibdex/github-app-token@3beb63f4bd073e61482598c45c71c1019b59b73a # v2.1.0
+        with:
+          app_id: ${{ secrets.TERRAFORM_AUTOMATION_APP_ID }}
+          private_key: ${{ secrets.TERRAFORM_AUTOMATION_PRIVATE_KEY }}
+      
+      - name: Checkout Template
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ steps.auth.outputs.token }}
+      
+      - name: Checkout Downstream
+        uses: actions/checkout@v4
+        with:
+          repository: workleap/${{ matrix.repo }}
+          token: ${{ steps.auth.outputs.token }}
+          path: downstream
+
+      - name: Set git author
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+      
+      - name: Merge Template Changes
+        shell: pwsh
+        run: |
+          $ErrorActionPreference = 'Stop'
+
+          Set-Location downstream
+          git remote add template "https://x-access-token:${{ steps.auth.outputs.token }}@github.com/workleap/${{ inputs.templateRepoName }}.git"
+          git fetch template
+
+          $tmpDir = ".tmp-sync"
+          if (-Not (Test-Path $tmpDir)) { New-Item -Path $tmpDir -ItemType Directory -Force | Out-Null }
+          git --work-tree=$tmpDir checkout template/main -- .github CONTRIBUTING.md SECURITY.md renovate.json
+          Remove-Item -Path "$tmpDir/.github/workflows/${{ inputs.workflowFileName }}" -Force
+
+          Copy-Item -Path "$tmpDir/.github" -Destination "." -Recurse -Force -ErrorAction Ignore
+          Copy-Item -Path "$tmpDir/CONTRIBUTING.md" -Destination "." -Force -ErrorAction Ignore
+          Copy-Item -Path "$tmpDir/SECURITY.md" -Destination "." -Force -ErrorAction Ignore
+          Copy-Item -Path "$tmpDir/renovate.json" -Destination "." -Force -ErrorAction Ignore
+          Remove-Item -Path $tmpDir -Recurse -Force
```

Comment:
I think you can just use the following command instead of creating a worktree

````
git checkout template/main -- CONTRIBUTING.md SECURITY.md renovate.json .github/**
````

Reply:
I would go as far as suggesting
```
git restore --source=template/main -- CONTRIBUTING.md SECURITY.md renovate.json .github/**
```
since I'm a git checkout hater now